src/VirtoCommerce.MarketingModule.Data/Repositories/IMarketingRepository.cs

```diff
@@ -36,5 +36,7 @@ public interface IMarketingRepository : IRepository
 
         Task<PromotionUsageEntity[]> GetMarketingUsagesByIdsAsync(string[] ids);
         Task RemoveMarketingUsagesAsync(string[] ids);
+
+        Task<string[]> CheckCouponsForUniqueness(Coupon[] coupons);
```

Comment:
Also please add Async suffix to the name

Reply:
We don't use await so we shouldn't use async.

Reply:
I meant Async suffix in the name, not the `async` keyword.
Looking at the method implementation, I think we need to add the `async` keyword and use awaitable `.ToArrayAsync()`. This method queues the data, and these actions could be asynchronous.